code/TagField.php

```diff
@@ -173,4 +273,90 @@ public function saveInto(DataObjectInterface $record) {
 			$record->$name = implode(',', $values);
 		}
 	}
+
+	/**
+	 * Returns a JSON string of tags, for ajax-based search.
+	 *
+	 * @param SS_HTTPRequest $request
+	 *
+	 * @return string
+	 */
+	public function suggest(SS_HTTPRequest $request) {
+		if($this->recordClass === null) {
+			return '[]';
+		}
+
+		/**
+		 * @var DataObject $object
+		 */
+		$object = singleton($this->recordClass);
```

Comment:
I added it to help with static analysis. `singleton` doesn't type-hint a class with `hasMethod` or `hasField` so a static analyser can't check expected types on that. Do you foresee a need to leave sections of this commented out? Perhaps that points to unnecessary code we could trim away?